src/platform/utilities/sso/index.js

```diff
@@ -44,16 +44,17 @@ export async function ssoKeepAliveSession() {
 }
 
 export async function checkAutoSession(application = null, to = null) {
-  const [{ ttl, authn }, hasSession] = await Promise.all([
+  const [{ ttl, authn }, userProfile] = await Promise.all([
     ssoKeepAliveSession(),
-    hasVAGovSession(),
+    vaGovProfile(),
   ]);
-  if (hasSession && ttl === 0) {
+  // FIXME: will a 3rd party lib auto convert the "sign_in" snake case to camel?
+  if (userProfile?.sign_in?.ssoe && ttl === 0) {
```

Comment:
I believe rails does this. if you look at a user request payload instead of navigating to the api page, it shows up as `signIn`

Reply:
I just tested on staging and the values that come back from the API are snake case.
<img width="533" alt="Screen Shot 2020-07-02 at 09 49 36" src="https://user-images.githubusercontent.com/170002/86388306-6b79cb80-bc49-11ea-8327-290c84c27f36.png">

Reply:
nvm, you're right. it's transformed in the [profile reducer](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/user/profile/utilities/index.js#L66)

Reply:
I think it's ok to leave it as is here, since we're only concerned about a single property

Reply:
vets-website passes an x-key-inflection header that tells vets-api to camelCase the keys.

Reply:
Using the olive_branch gem.